src/refract/element.es6

```diff
@@ -0,0 +1,92 @@
+/*
+ * Base elements for working with refract shorthand, which is composed of
+ * tuples with the following values:
+ *
+ * [name, metadata, attributes, contents]
+ *
+ * For example:
+ *
+ * ['resource', {'title': 'Question'}, {'href': '/questions'},
+ *   ['transition', {}, {}, ...],
+ *   ...
+ * ]
+ *
+ * https://github.com/refractproject/refract-spec
+ */
+const rawElement = Symbol('element');
+
+export default class RefractElement {
```

Comment:
Perhaps it would make sense to use @smizell's [minim](https://github.com/smizell/minim) library.